### DISCUSSION TITLE
rename styles?

### DISCUSSION THREAD

**Discussion by wordmuse** [freeplane/freeplane#441] [Help]
Is there any way to rename a style? It seems like there ought to be a way, but I'm not seeing it. Am I missing something obvious (wouldn't be the first time)?

~ Bal

**Comment by abc16361** [comment]
If it changes, how about those nodes and scripts using it?

**Comment by dpolivaev** [comment]
Unfortunately, style names are used as style ids across the whole map. Therefore there is no easy way to change them.

**Reply by dpolivaev** [reply]
To be precise, you can not easily change them inside Freeplane. But you could easily change them using a text editor. However if the same words are used also inside your content you need to be careful.

**Reply by wordmuse** [reply]
OK - thanks Dimitri. I'll create a test copy of my mindmap and work with it.
~ Bal

**Comment by EdoFro** [comment]
Y had to do something like that a time ago.

The steps to rename (or replace) a style are:

1. Create a new style from the style you want to rename and name it with the new name:
![image](https://user-images.githubusercontent.com/35700575/172956976-1a7055b7-0d30-4961-ab57-e154ab54c9d9.png)

    - select it and right click on it
 
2. Type the new name:
![image](https://user-images.githubusercontent.com/35700575/172957217-e44a3ffe-93b3-46ac-8f3d-800ac35c8c2c.png)

now it looks like this:
![image](https://user-images.githubusercontent.com/35700575/172957313-cf7bf462-a964-48b0-a541-744c0e2e1d2d.png)

3. then return to the map and run the following script. If you want to change the styles in the whole map then select the root node. You can select another node if you want to change the styles from its descendants only.

```groovy
//here you can add all styles you want to change/rename
def dict = [
    'old style name'     : 'new style name'
    ,'Rule_1'   : 'myNewName'
]

def nodo = node //or node.map.root if you want the whole map
def texto = new StringBuilder()

dict.each{ k, v ->
    def nodos = nodo.find{it.style.name == k}
    nodos*.style*.name = v
    texto << "\n  ${nodos.size()} nodes  with style '$k'   changed to style   '$v'"
}
ui.informationMessage(texto.toString())
```

4. if you have conditional styles, you can change them in the conditional style manager
    - change the old name to the new name

5. use the filter to see if there are any node with the old style
6. if not, use the style editor to delete the old style.

hope it helps,

edo

**Reply by EdoFro** [reply]
I changed the color of the new style, so you can see in the images what happens when I apply the script:
![image](https://user-images.githubusercontent.com/35700575/172959041-5023ea6f-b866-4ec9-9bd3-d8d082fa6702.png)

before applying the script:
![image](https://user-images.githubusercontent.com/35700575/172959134-041e24da-f7d3-420e-9fb5-11a34fc87b32.png)

after applying the script:
![image](https://user-images.githubusercontent.com/35700575/172959224-64b3a39d-57e8-445c-a0ce-6a511f8faabd.png)

**Comment by wordmuse** [comment]
Thanks everyone. I think Dimitri's idea of working with a text editor will get me to where I want to be. I will, of course, do the changes in a copy of the original mindmap so that I screw up anything, the original is still in good shape.

I appreciate the time you folks have put into giving me an answer.

~ Bal

**Comment by euu2021** [comment]
[Another solution](https://sourceforge.net/p/freeplane/featurerequests/2597/#195c):

(let's say you want to replace the name Style A for Style B):

-in the Style Editor, create Style B by right clicking Style A and then "new user style from selection"
-Set the Quick Filter bar to "Style is equal to Style A". Don't apply the filter.
-Filter->Select all matching nodes
-Format->Apply style->Style B

**Reply by EdoFro** [reply]
@eu2021, you are right. This is the way for normal use.

I used the script shown in the example [above](https://github.com/freeplane/freeplane/discussions/441#discussioncomment-2917737)  because I had several styles to rename (translate them from spanish to english) in multiple maps.
